app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -66,4 +69,19 @@ protected void log(int priority, String tag, String message, Throwable throwable
     public DaoSession getDaoSession() {
         return daoSession;
     }
+
+    /**
+     * Initializes Retrofit and makes it available to all activities.
+     */
+    public Retrofit getRetrofit() {
+        Timber.i("getRetrofit");
```

Comment:
Retrofit should be initialized as singleton and used across threads because creating a Retrofit instance each time a request is executed is inefficient in terms of performance.

Reply:
It's only initalized once during the lifetime of the application?

```
if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BuildConfig.REST_URL + "/")
                    .build();
        }
```

Is it okay like this?

Reply:
I think it's fine enough, sorry, you don't need to change this.